src/libraries/System.Net.Sockets/tests/FunctionalTests/UdpClientTest.cs

```diff
@@ -623,6 +672,23 @@ public async Task SendAsync_ReceiveAsync_Connected_Success()
             }
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)] // "localhost" resolves to IPv4 & IPV6 on Windows, but may resolve to only one of those on Unix
+        [OuterLoop]
+        public async ValueTask SendAsyncWithReadOnlyMemory_ReceiveAsync_Connected_Success()
+        {
+            using (var receiver = new UdpClient("localhost", 0))
+            using (var sender = new UdpClient("localhost", ((IPEndPoint)receiver.Client.LocalEndPoint).Port))
+            {
+                await sender.SendAsync(new ReadOnlyMemory<byte>(new byte[1]));
```

Comment:
Same as in `SendAsyncWithReadOnlyMemory_ReceiveAsync_Success`: we need to test the other overloads too. We also need to verify if `InvalidOperationException` is being thrown on `hostName != null` or `port != 0` for a connected `UdpClient`.

Reply:
Added.